### PR TITLE
feat: rename ThemeProvider to OrbitProvider

### PR DESCRIPTION
**Pull request by DSil**
[ORBIT-1940](https://jira.kiwi.com/browse/ORBIT-1940)

BREAKING CHANGE: ThemeProvider is now called OrbitProvider.

All references to ThemeProvider were replaced and a new documentation page was also created for the provider.
 Storybook: https://orbit-mainframev-feat-rename-theme-provider.surge.sh

### CODE REVIEW

File: docs/src/documentation/03-components/14-utility/orbitprovider/01-guidelines.mdx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Guidelines
+redirect_from:
+  - /utilities/orbit-provider/
+---
+
+## When to use
+
+- To have Orbit [theming](/foundation/color/theming/) and [dictionary](/development/guides/dictionary/) on your app.
```
Comment (mainframev) [inline docs/src/documentation/03-components/14-utility/orbitprovider/01-guidelines.mdx:9]: I think this tab is not needed. Let's left only `react` tab ;)

**Review by mainframev — CHANGES_REQUESTED** [review]
LGTM

**Review by mainframev — APPROVED** [review]

**Comment by mainframev** [comment]
let's merge it later, when other breaking changes will be ready :)

**Comment by DSil** [comment]
This will require an update on the orbit-themer package after #3615 is merged

**Pull request merged**